diffkemp/simpll/DifferentialFunctionComparator.cpp

```diff
@@ -847,31 +708,20 @@ int DifferentialFunctionComparator::cmpBasicBlocks(
             if (Res) {
                 // The instructions are indeed different, try to find the source
                 // of the difference.
-
-                // Try to find macros that could be causing the difference
-                auto macroDiffs =
-                        ModComparator->MacroDiffs.findMacroDifferences(&*InstL,
-                                                                       &*InstR);
-                ModComparator->ComparedFuns.at({FnL, FnR})
-                        .addDifferingObjects(std::move(macroDiffs));
-
-                // Try to find assembly functions causing the difference
-                if (isa<CallInst>(&*InstL) && isa<CallInst>(&*InstR)
-                    && config.PrintAsmDiffs) {
-                    ModComparator->ComparedFuns.at({FnL, FnR})
-                            .addDifferingObjects(findAsmDifference(
-                                    dyn_cast<CallInst>(&*InstL),
-                                    dyn_cast<CallInst>(&*InstR)));
-                }
-
-                if (isa<CallInst>(&*InstL) && isa<CallInst>(&*InstR))
-                    processCallInstDifference(dyn_cast<CallInst>(&*InstL),
-                                              dyn_cast<CallInst>(&*InstR));
-
+                findDifference(&*InstL, &*InstR);
                 return Res;
             }
         }
 
+        if (isa<CallInst>(&*InstL) && isa<CallInst>(&*InstR)) {
+            // Try to find a type difference in field accesses.
+            // This has to be done even if the operations are equal, since field
+            // accesses return a pointer and all pointers are compared as equal
+            // in FunctionComparator.
+            findTypeDifferenceInFieldAccess(dyn_cast<CallInst>(&*InstL),
```

Comment:
I think it should be clearly stated here that this handles chained field abstractions (both in the comment and in the function name) to distinguish it from the case handled in `processCallInstDifference` by calling `findTypeDifferences` directly (which also looks for "a type difference in field accesses").